### DISCUSSION TITLE
Dispose of Avalonia.Media.Imaging.Bitmap?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#6917] [Q&A] [answered]
Hi there,

I'd like to generate some content at runtime and assign it to Image.Source. So I'm look at Bitmap and, in particular, RenderTargetBitmap.

However, Bitmap is disposable. I look for examples online, but no examples I've seen actually dispose of Bitmap instances.

If I assign a Bitmap to Image.Source, do I need to dispose of the Bitmap instance?

Should I be assigning something else?

**Comment by kekekeks** [comment, marked as answer]
>I look for examples online, but no examples I've seen actually dispose of Bitmap instances.

WPF's bitmaps are not disposable, so lots of examples rely on GC too. If you are dealing with lots of images loaded dynamically you need to dispose bitmaps explicitly

>If I assign a Bitmap to Image.Source, do I need to dispose of the Bitmap instance?

Yes, if you are dealing with lots of bitmaps. GC will take care if it automatically otherwise.

>Should I be assigning something else?

Set `Image.Source` to `null` before disposing the bitmap

**Reply by kuiperzone** [reply]
OK. Thank you.
